nilearn/glm/tests/test_first_level.py

```diff
@@ -2235,24 +2223,47 @@ def test_flm_with_surface_masker_with_mask(_make_surface_glm_data, surf_mask):
     assert isinstance(model.masker_, SurfaceMasker)
 
 
-def test_flm_with_surface_data_no_design_matrix(_make_surface_glm_data):
+def test_flm_with_surface_data_no_design_matrix(surface_glm_data):
     """Smoke test FirstLevelModel with surface data and no design matrix."""
-    img, _ = _make_surface_glm_data(5)
+    img, _ = surface_glm_data(5)
     masker = SurfaceMasker().fit(img)
     model = FirstLevelModel(mask_img=masker, t_r=2.0)
     model.fit(img, events=basic_paradigm())
 
 
-def test_flm_compute_contrast_with_surface_data(_make_surface_glm_data):
+def test_flm_compute_contrast_with_surface_data(surface_glm_data):
     """Smoke test FirstLevelModel compute_contrast with surface data."""
-    img, _ = _make_surface_glm_data(5)
+    img, _ = surface_glm_data(5)
     masker = SurfaceMasker().fit(img)
     model = FirstLevelModel(mask_img=masker, t_r=2.0)
     events = basic_paradigm()
     model.fit([img, img], events=[events, events])
     result = model.compute_contrast("c0")
 
     assert isinstance(result, SurfaceImage)
+    assert_polymesh_equal(img.mesh, result.mesh)
+
+
+def test_flm_get_voxelwise_model_attribute_with_surface_data(
+    surface_glm_data,
+):
+    """Smoke test 'voxel wise' attribute with surface data.
+
+    TODO: rename the private function _get_voxelwise_model_attribute
+    to work for both voxel and vertex
+    """
+    img, _ = surface_glm_data(5)
+    masker = SurfaceMasker().fit(img)
+    model = FirstLevelModel(mask_img=masker, t_r=2.0, minimize_memory=False)
+    events = basic_paradigm()
+    model.fit([img, img], events=[events, events])
+
+    model.residuals  # noqa: B018
+    model.predicted  # noqa: B018
+    model.r_square  # noqa: B018
```

Comment:
Should we be checking something here with `assert` statements?

Reply:
fixed